### PR TITLE
Support date format in locale

### PR DESCRIPTION
**Pull request by zombieJ**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

#13196
  
### API Realization (Optional if not new feature)

```jsx
const myLocale = {
    DatePicker: {
      // ...
      dateFormat: 'YYYY 年 M 月 D 日',
      dateTimeFormat: 'YYYY 年 M 月 D 日 H 时 m 分 s 秒',
      weekFormat: 'YYYY 年 W 周',
      monthFormat: 'YYYY 年 M 月',
    },
  };

<LocaleProvider locale={myLocale}>
  <Picker value={date} {...props} />
</LocaleProvider>
```

### What's the effect? (Optional if not new feature)

* Not affect current usage.
* Affect future default format when social contributes the default locale format with LocaleProvider.

### Changelog description (Optional if not new feature)

* DatePicker now will read format as default format  in locale if provided.
* DatePicker 将会读取本地化格式配置作为默认日期格式。

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

Need social contributors PR to add locale format in locale files.

### CODE REVIEW

File: components/date-picker/wrapPicker.tsx [hunk 1]
```diff
@@ -6,6 +6,25 @@ import LocaleReceiver from '../locale-provider/LocaleReceiver';
...
+  date: 'YYYY-MM-DD',
+  week: 'gggg-wo',
+  month: 'YYYY-MM',
+};
+
+const LOCALE_FORMAT_MAPPING: PickerMap = {
+  date: 'dateFormat',
+  dateTime: 'dateTimeFormat',
+  week: 'weekFormat',
+  month: 'monthFormat',
```
Comment (afc163) [inline components/date-picker/wrapPicker.tsx]: monthFormat 不是不能用么

Comment (zombieJ) [reply components/date-picker/wrapPicker.tsx]: 原本的 format 都在 lang 里面，现在在 lang 平级加了一个 `xxxFormat`， lang 里面的不动它。

Comment (afc163) [reply components/date-picker/wrapPicker.tsx]: 👌

**Comment by afc163** [comment]
`What's the effect?` 可以补充下可能的预期外变化。

**Comment by zombieJ** [comment]
> `What's the effect?` 可以补充下可能的预期外变化。

👌

**Review by afc163 — APPROVED** [review]

**Comment by afc163** [comment]
记得删分支

**Pull request merged**